datashader/resampling.py

```diff
@@ -79,30 +84,241 @@
+    ychunks = list(range(0, outy, cys)) + [outy]
+    iny, inx = in_shape
+    xscale = inx/outx
+    yscale = iny/outy
+    mapping = {}
+    for i in range(len(ychunks)-1):
+        cumy0, cumy1 = ychunks[i:i+2]
+        iny0, iny1 = cumy0*yscale, cumy1*yscale
+        iny0r, iny1r = floor(iny0), ceil(iny1)
+        y0_off, y1_off = iny0-iny0r, iny1r-iny1
+        for j in range(len(xchunks)-1):
+            cumx0, cumx1 = xchunks[j:j+2]
+            inx0, inx1 = cumx0*xscale, cumx1*xscale
+            inx0r, inx1r = floor(inx0), ceil(inx1)
+            x0_off, x1_off = inx0-inx0r, inx1r-inx1
+            mapping[(i, j)] = {
+                'out': {
+                    'x': (cumx0, cumx1),
+                    'y': (cumy0, cumy1),
+                    'w': (cumx1-cumx0),
+                    'h': (cumy1-cumy0),
+                },
+                'in': {
+                    'x': (inx0r, inx1r),
+                    'y': (iny0r, iny1r),
+                    'xoffset': (x0_off, x1_off),
+                    'yoffset': (y0_off, y1_off),
+                }
+            }
+    return mapping
+
+
+def compute_chunksize(src, w, h, chunksize=None, max_mem=None):
+    """
+    Attempts to compute a chunksize for the resampling output array
+    that is as close as possible to the input array chunksize, while
+    also respecting the maximum memory constraint to avoid loading
+    to much data into memory at the same time.
+
+    Parameters
+    ----------
+    src : dask.array.Array
+        The source array to resample
+    w : int
+        New grid width
+    h : int
+        New grid height
+    chunksize : tuple(int, int) (optional)
+        Size of the output chunks. By default this the chunk size is
+        inherited from the *src* array.
+    max_mem : int (optional)
+        The maximum number of bytes that should be loaded into memory
+        during the regridding operation.
+
+    Returns
+    -------
+    chunksize : tuple(int, int)
+        Size of the output chunks.
+    """
+    start_chunksize = src.chunksize if chunksize is None else chunksize
+    if max_mem is None:
+        return start_chunksize
+
+    sh, sw = src.shape
+    height_fraction = float(sh)/h
+    width_fraction = float(sw)/w
+    ch, cw = start_chunksize
+    dim = True
+    nbytes = src.dtype.itemsize
+    while ((ch * height_fraction) * (cw * width_fraction) * nbytes) > max_mem:
+        if dim:
+            cw -= 1
+        else:
+            ch -= 1
+        dim = not dim
+    if ch == 0 or cw == 0:
+        min_mem = height_fraction * width_fraction * nbytes
+        raise ValueError(
+            "Given the memory constraints the resampling operation "
+            "could not find a chunksize that avoids loading too much "
+            "data into memory. Either relax the memory constraint to "
+            "a minimum of %d bytes or resample to a larger grid size. "
+            "Note: A future implementation may handle this condition "
```

Comment:
```suggestion
            "Note: A future implementation could handle this condition "
```

(Changed to a message that will prompt someone to write a PR if they need this, rather than promising that we might do this ourselves. :-)